File: hello-kotlin/src/main/kotlin/client/util/StreamIdGenerator.kt
package client.util

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import kotlin.random.Random

/**
 * Stream ID generation utilities
 */
object StreamIdGenerator {
    private val formatter = DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss")
    
    /**
     * Generate a unique stream ID
     */
    fun generate(): String {
        val timestamp = LocalDateTime.now().format(formatter)
        val random = Random.nextInt(0, 0xFFFFFF).toString(16).padStart(8, '0')
        return "stream-$timestamp-$random"
    }
}

File: hello-swift/Sources/Client/Util/StreamIdGenerator.swift
import Foundation

/// Stream ID generator for creating unique stream identifiers
class StreamIdGenerator {
    /// Generate a unique stream ID with timestamp and random component
    static func generate() -> String {
        let dateFormatter = DateFormatter()
        dateFormatter.dateFormat = "yyyyMMdd-HHmmss"
        let timestamp = dateFormatter.string(from: Date())
        let random = String(format: "%08x", Int.random(in: 0...0xFFFFFFFF))
        return "stream-\(timestamp)-\(random)"
    }
    
    /// Validate stream ID format
    static func validate(_ streamId: String) -> Bool {
        return streamId.hasPrefix("stream-") && streamId.count > 8
    }
}
